### PR TITLE
ruffle: nightly-2023-04-10 -> nightly-2024-02-09

### PR DESCRIPTION
**Pull request by jchv**
## Description of changes

Updates Ruffle from nightly-2023-04-10 to ~~nightly-2024-01-21~~ ~~nightly-2024-01-27~~ nightly-2024-02-09.

There are absolutely *tons* of changes, improvements, bugfixes, etc. since the last nightly in Nixpkgs. Ruffle now has somewhat of a proper "GUI" including some basic debugging functionality. ActionScript 3 compatibility continues to improve as well. For more information on what's been going on in Ruffle lately, check out their [latest blog post](https://ruffle.rs/blog/2024/01/14/2023-in-review).

On the nixpkgs side of things, we can now remove the terrible hack needed to work around #183344 since there is no longer a dependency on `dasp_sample` from git. Also, this is not strictly related to the version update, but I've added `ruffle_desktop` as the `mainProgram` so that `nix run` works.

Adding myself as a maintainer, since I have been the only person updating the Ruffle derivation for a couple years now.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/emulators/ruffle/default.nix [hunk 1]
```diff
@@ -19,13 +19,13 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "ruffle";
-  version = "nightly-2023-04-10";
+  version = "nightly-2024-01-21";
 
   src = fetchFromGitHub {
     owner = "ruffle-rs";
     repo = pname;
```
Comment (dotlambda) [inline pkgs/applications/emulators/ruffle/default.nix:26]:
```suggestion
    repo = "ruffle";
```
See https://github.com/NixOS/nixpkgs/issues/277994.

File: pkgs/applications/emulators/ruffle/default.nix [hunk 2]
```diff
@@ -19,13 +19,13 @@
...
     rev = version;
-    sha256 = "sha256-u5Ri9KnYzE3JedUP9fGgYeG8G9uxrL6/zt3KPiKjhU0=";
+    sha256 = "sha256-CcjG8Gtwdrs3j6AnYiUbM+gcRnSUmPpLx/yNfu2JJqo=";
```
Comment (dotlambda) [inline pkgs/applications/emulators/ruffle/default.nix:28]:
```suggestion
    hash = "sha256-CcjG8Gtwdrs3j6AnYiUbM+gcRnSUmPpLx/yNfu2JJqo=";
```

**Review by dotlambda — COMMENTED** [review]
`meta.description` must not end in a period.

**Comment by jchv** [comment]
Rebased + updated to nightly 2024-01-27 (which removes another Git dependency.)

File: pkgs/applications/emulators/ruffle/default.nix [hunk 3]
```diff
@@ -73,29 +73,24 @@ rustPlatform.buildRustPackage rec {
...
+      "h263-rs-0.1.0" = "sha256-Akf1SBjo8qikhiHI8NPvO3vJvVfm0dQBf2X9V7OdgQc=";
+      "jpegxr-0.3.0" = "sha256-jirUbse2MiUDCmwBO7ykWNKHgDgL/6ZM5o2HeDUhm0c=";
       "nellymoser-rs-0.1.2" = "sha256-GykDQc1XwySOqfxW/OcSxkKCFJyVmwSLy/CEBcwcZJs=";
-      "nihav_codec_support-0.1.0" = "sha256-rE9AIiQr+PnHC9xfDQULndSfFHSX4sqKkCAQYVNaJcQ=";
+      "nihav_codec_support-0.1.0" = "sha256-HAJS4I6yyzQzCf+vmaFp1MWXpcUgFAHPxLhfMVXmN1c=";
     };
   };
 
   meta = with lib; {
     description = "An Adobe Flash Player emulator written in the Rust programming language.";
```
Comment (dotlambda) [inline pkgs/applications/emulators/ruffle/default.nix:89]:
```suggestion
    description = "An Adobe Flash Player emulator written in the Rust programming language";
```

Comment (jchv) [reply pkgs/applications/emulators/ruffle/default.nix]: Sorry, I did actually fix this but apparently I must've reverted it by accident somehow.

**Comment by edanaher** [comment]
I tried using this branch to build a newer ruffle_desktop and it built fine, but I got an error on startup: `Library libxkbcommon-x11.so could not be loaded.`.  AFAICT, this is dynamically loaded at runtime but not used at all in the build process.

Adding a dependency on libxkbcommon along with the following patchelf line fixed it for me.
```
  preFixup = ''
    ${patchelf}/bin/patchelf $out/bin/ruffle_desktop --add-needed libxkbcommon-x11.so --add-rpath ${libxkbcommon}/lib
  '';
```
I'm not sure if this is me doing something weird or just weirdness in Ruffle; I assume that "Tested basic functionality of all binary files" would have caught this before submitting the PR...

**Comment by jchv** [comment]
Ahhh, I can explain: All of my machines are Wayland-based, so I simply didn't notice. If I run `WAYLAND_DISPLAY= nix run .#ruffle` I can immediately reproduce when running under XWayland. Sorry, I'll have to remember that next time I update Ruffle.

I'll take a look at this on the weekend.

**Comment by jchv** [comment]
So much for "the weekend." But alas, I did look at it, and indeed, that fix seems to be the way to go, so I applied it and also updated the PR to `nightly-2024-02-09` for good measure, removing another Git dependency from `Cargo.lock`.

**Pull request merged**